service/src/main/java/uk/gov/mca/beacons/api/duplicates/application/DuplicatesService.java

```diff
@@ -0,0 +1,122 @@
+package uk.gov.mca.beacons.api.duplicates.application;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.mca.beacons.api.beacon.application.BeaconService;
+import uk.gov.mca.beacons.api.beacon.domain.Beacon;
+import uk.gov.mca.beacons.api.duplicates.domain.DuplicateBeacon;
+import uk.gov.mca.beacons.api.duplicates.domain.DuplicatesRepository;
+import uk.gov.mca.beacons.api.duplicates.domain.DuplicatesSummary;
+import uk.gov.mca.beacons.api.duplicates.rest.DuplicateBeaconDTO;
+import uk.gov.mca.beacons.api.exceptions.ResourceNotFoundException;
+import uk.gov.mca.beacons.api.legacybeacon.application.LegacyBeaconService;
+import uk.gov.mca.beacons.api.legacybeacon.domain.LegacyBeacon;
+
+@Service("DuplicatesService")
+public class DuplicatesService {
+
+  private final DuplicatesRepository duplicatesRepository;
+  private final BeaconService beaconService;
+  private final LegacyBeaconService legacyBeaconService;
+
+  @Autowired
+  public DuplicatesService(
+    DuplicatesRepository duplicatesRepository,
+    BeaconService beaconService,
+    LegacyBeaconService legacyBeaconService
+  ) {
+    this.duplicatesRepository = duplicatesRepository;
+    this.beaconService = beaconService;
+    this.legacyBeaconService = legacyBeaconService;
+  }
+
+  public List<DuplicatesSummary> getDuplicateSummaries(
+    int pageNumber,
+    int numberPerPage
+  ) {
+    int numberAlreadyTaken = pageNumber * numberPerPage;
+
+    List<DuplicatesSummary> duplicateSummaries = getBatch(
+      numberPerPage,
+      numberAlreadyTaken
+    );
+
+    return duplicateSummaries;
+  }
+
+  private List<DuplicatesSummary> getBatch(
+    int numberPerPage,
+    int numberAlreadyTaken
+  ) {
+    return duplicatesRepository
+      .findAll()
+      .stream()
+      .skip(numberAlreadyTaken)
+      .limit(numberPerPage)
+      .collect(Collectors.toList());
+  }
+
+  public List<DuplicateBeaconDTO> getDuplicatesForHexId(String hexId) {
+    List<DuplicateBeacon> duplicateBeaconsForHexId = new ArrayList<>();
+    List<DuplicateBeaconDTO> duplicateBeaconDTOs = new ArrayList<>();
+
+    try {
+      List<Beacon> duplicateModernBeacons = beaconService.findByHexId(hexId);
```

Comment:
This implies that if a hexID exists in modern, it won't exist in legacy. 

Real life scenario is more likely:

They had a legacy registration, now they have a modern one.